src/app/pages/mini-games/find-pokemon-name/game/layout/InteractionScreen.jsx

```diff
@@ -1,11 +1,33 @@
-import { useContext } from 'react';
+import { useCallback, useContext, useEffect } from 'react';
 import SelectLetter from '../components/SelectLetter';
 import { FindPokemonNameContext } from '../state/Context';
 
 const InteractionScreen = () => {
   const { state, startGame } = useContext(FindPokemonNameContext);
   const { pokemon, gameStatus } = state;
 
+  const onKeyDown = useCallback(
+    (e) => {
+      if (gameStatus !== 'win' && gameStatus !== 'lost') {
+        return;
+      }
+
+      const { code } = e;
+
+      if (code !== 'Enter' && code !== 'Space') {
+        return;
+      }
+
+      startGame();
+    },
+    [gameStatus, startGame],
+  );
+
+  useEffect(() => {
+    window.addEventListener('keydown', onKeyDown, false);
+    return () => window.removeEventListener('keydown', onKeyDown, false);
+  }, [onKeyDown]);
```

Comment:
Maybe now that there is more logic, move the selected code together with this: 
```
  if (gameStatus === 'win') {
    return <button onClick={startGame}>NEXT POKEMON</button>;
  }
  if (gameStatus === 'lost') {
    return <button onClick={startGame}>TRY AGAIN</button>;
  }
```

to a separate component inside the `mini-games/find-pokemon-name/game/components` folder. 
In this way, onKeyDown will not fire at all. Then you will remove the early return from the onKeyDown. 

So the template it will go from this:
```
  if (!pokemon?.name) {
    return null;
  }
  if (gameStatus === 'win') {
    return <button onClick={startGame}>NEXT POKEMON</button>;
  }
  if (gameStatus === 'lost') {
    return <button onClick={startGame}>TRY AGAIN</button>;
  }
  return <SelectLetter />;
```

To something like this:

```
  if (!pokemon?.name) {
    return null;
  }
  if (gameStatus === 'win' || gameStatus === 'lost') {
    return <RoundCompleted />; // The name is not final add whatever you think is more correct.
  }
  return <SelectLetter />;
```

Reply:
Thanks for the feedback, it's much cleaner like this! I ended up naming it `RoundResult`. Please let me know what do you think!